src/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -1234,6 +1272,32 @@ your own zpool.\n"
 
 	promptpool
 
+	# Set value for BOOTPOOL now.  If the user created one or more
+	# additional non-zones pool(s) manually during the promptpool
+	# function, use the first non-zones one we find.
+	BOOTPOOL=$(zpool list -Ho name | grep -vw zones | head -1)
+	if [[ "$BOOTPOOL" == "" ]]; then
+		BOOTPOOL="zones"
+	fi
+
+	printheader "Self-booting"
+
+	message="
+SmartOS can boot off either the \"zones\" zpool, or a dedicated named
+zpool in lieu of a USB stick or a CD-ROM.  Enter 'yes' if you wish to try and
+make a SmartOS zpool self-booting.\n"
+
+	
+	if [[ $(getanswer "skip_instructions") != "true" ]]; then
+	    printf "$message"
+	fi
+
+	promptyesno "Boot SmartOS from a zpool" $boot_from_zpool
```

Comment:
This still seems to be forcing interactive usage?

Reply:
$boot_from_zpool is set to "no" at the very top, so just hitting RETURN here will answer "no".  If "hitting RETURN a new time" is "forcing interactive usage" then yes, I'm adding one (two if you answer "yes") more question.

Reply:
Please check the rest of the script: it should be possible to supply a config file such that it automatically installs. That's what you're breaking by requiring any input. Look at getanswer() is you're still not clear